packages/frontend/editor-ui/src/features/ai/chatHub/ChatView.test.ts

```diff
@@ -0,0 +1,625 @@
+					anthropic: {
+						models: [
+							createMockAgent({
+								name: 'Claude 3',
+								model: { provider: 'anthropic', model: 'claude-3' },
+							}),
+						],
+					},
+				}),
+			);
+
+			const rendered = renderComponent({ pinia });
+
+			expect(await rendered.findByRole('button', { name: /GPT-4/ })).toBeInTheDocument();
+		});
+
+		it('should show callout if no agent is available', async () => {
+			vi.mocked(chatApi.fetchChatModelsApi).mockResolvedValue(emptyChatModelsResponse);
+
+			const rendered = renderComponent({ pinia });
+
+			expect(await rendered.findByText('select a model')).toBeInTheDocument();
+			expect(await rendered.findByRole('textbox')).toBeDisabled();
+		});
+	});
+
+	describe('Rendering existing sessions', () => {
+		beforeEach(() => {
+			mockRoute.params = { id: 'existing-session-123' };
+
+			vi.mocked(chatApi.fetchSingleConversationApi).mockResolvedValue(
+				createMockConversationResponse({
+					session: createMockSession({
+						id: 'existing-session-123',
+						title: 'Test Conversation',
+						lastMessageAt: new Date().toISOString(),
+						provider: 'custom-agent',
+						agentId: 'agent-123',
+					}),
+					conversation: {
+						messages: {
+							'msg-1': createMockMessageDto({
+								id: 'msg-1',
+								sessionId: 'existing-session-123',
+								content: 'What is the weather today?',
+							}),
+							'msg-2': createMockMessageDto({
+								id: 'msg-2',
+								sessionId: 'existing-session-123',
+								type: 'ai',
+								name: 'Assistant',
+								content: 'The weather is sunny today.',
+								provider: 'custom-agent',
+								agentId: 'agent-123',
+								previousMessageId: 'msg-1',
+							}),
+						},
+					},
+				}),
+			);
+		});
+
+		it('preselects agent set for the session', async () => {
+			const rendered = renderComponent({ pinia });
+
+			expect(
+				await rendered.findByRole('button', { name: /Test Custom Agent/i }),
+			).toBeInTheDocument();
+		});
+
+		it('displays conversation with messages loaded from API', async () => {
+			const rendered = renderComponent({ pinia });
+
+			await vi.waitFor(() => {
+				const messages = rendered.container.querySelectorAll('[data-message-id]');
+
+				expect(messages).toHaveLength(2);
+				expect(messages[0]).toHaveTextContent('What is the weather today?');
+				expect(messages[1]).toHaveTextContent('The weather is sunny today.');
+			});
+		});
+
+		it('displays error toast and redirects to chat view when fetching conversation fails', async () => {
+			vi.mocked(chatApi.fetchSingleConversationApi).mockRejectedValue(
+				new Error('Conversation not found'),
+			);
+
+			const rendered = renderComponent({ pinia });
+
+			expect(await rendered.findByText(/Failed to load conversation/i)).toBeInTheDocument();
+
+			await vi.waitFor(() => expect(mockRouterPush).toHaveBeenCalledWith({ name: 'chat' }));
+		});
+
+		it.todo(
+			'handles when the agent selected for the conversation is not available anymore',
+			async () => {
```

Comment:
Realized this is broken because chatStore.getAgent() was changed to never return null. I'll make a fix in a separate PR.